src/implementations/vanilla/components/content-block/content-block.js

```diff
@@ -0,0 +1,152 @@
+import { queryOne } from '@ecl/dom-utils';
+
+/**
+ * @param {HTMLElement} element DOM element for component instantiation and scope
+ * @param {Object} options
+ * @param {Boolean} options.attachClickListener Whether or not to bind click events
+ * @param (String) options.dataSelector The selector of the element where to attach the click listener
+ * @param (String) options.dataTitle  The selector of the element containing the link
+ * @param (Integer) options.maxIterations Maximum number of ancestors to look for the element
+ */
+export class ContentBlock {
+  /**
+   * @static
+   * Shorthand for instance creation and initialisation.
+   *
+   * @param {HTMLElement} root DOM element for component instantiation and scope
+   *
+   * @return {ContentBlock} An instance of ContentBlock.
+   */
+  static autoInit(root, { CONTENT_BLOCK: defaultOptions = {} } = {}) {
+    const contentBlock = new ContentBlock(root, defaultOptions);
+    contentBlock.init();
+    root.ECLContentBlock = contentBlock;
+    return contentBlock;
+  }
+
+  constructor(
+    element,
+    {
+      dataSelector = '.ecl-picture',
```

Comment:
we could limit the risk of having bad element selected, by using a specific data attributes for pictures which could benefit from this.
For instance, adding `data-ecl-picture-click` only in the card and content item component